.github/workflows/test.yaml

```diff
@@ -47,9 +47,60 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v3
+      - name: Gather files to scan
+        shell: bash
+        id: gather
+        run: |
+          declare -a filepaths
+          shebangregex="^#! */[^ ]*/(env *)?[abk]*sh"
+
+          set -f # temporarily disable globbing so that globs in inputs aren't expanded
+
+          while IFS= read -r -d '' file; do
+          filepaths+=("$file")
+          done < <(find . \
+                        -type f \
+                        '(' \
+                        -name '*.bash' \
+                        -o -name '.bashrc' \
+                        -o -name 'bashrc' \
+                        -o -name '.bash_aliases' \
+                        -o -name '.bash_completion' \
+                        -o -name '.bash_login' \
+                        -o -name '.bash_logout' \
+                        -o -name '.bash_profile' \
+                        -o -name 'bash_profile' \
+                        -o -name '*.ksh' \
+                        -o -name 'suid_profile' \
+                        -o -name '*.zsh' \
+                        -o -name '.zlogin' \
+                        -o -name 'zlogin' \
+                        -o -name '.zlogout' \
+                        -o -name 'zlogout' \
+                        -o -name '.zprofile' \
+                        -o -name 'zprofile' \
+                        -o -name '.zsenv' \
+                        -o -name 'zsenv' \
+                        -o -name '.zshrc' \
+                        -o -name 'zshrc' \
+                        -o -name '*.sh' \
+                        -o -path '*/.profile' \
+                        -o -path '*/profile' \
+                        -o -name '*.shlib' \
+                        ')' \
+                        -print0)
+
+          while IFS= read -r -d '' file; do
+          head -n1 "$file" | grep -Eqs "$shebangregex" || continue
```

Comment:
I'm wondering if you couldn't use the `file` command that will probably be better at matching if it's a shell script

Reply:
That's good feedback - I dug in and specifically for shell scripts there is no difference, using `file` over a shell script will try to match a shebang as listed here https://github.com/file/file/blob/ba7b3172d658e3392ff7764c48dd2039c7ef026d/magic/Magdir/commands and if the file contains valid shell code and no shebang, `file` will categorize it as regular ASCII file. The behavior with `grep` and matching against the shebang is exactly the same.